src/forceUpdatedComponent.js

```diff
@@ -0,0 +1,41 @@
+import React, { PureComponent } from 'react'
+import { wrapDisplayName } from 'recompose'
+
+import { subscribe, unsubscribe } from './pubsub.js'
+
+/**
+ * Returns a HOC that triggers a re-render of itself whenever
+ * a subscription notification is received.
+ */
+export default function forceUpdatedComponent(InputComponent) {
+  class SubscribeUpdater extends PureComponent {
+    constructor(props) {
+      super(props)
+
+      this.state = {
+        lastNotification: null,
+      }
+    }
+
+    componentDidMount() {
+      subscribe(this.handleNotification.bind(this))
```

Comment:
Should the subscribe/unsubscribe be exposed via the context provider to support potentially multiple different translation provides within a single application?

Reply:
Let's look at that if the need arises!